resolver/type_syntax.h

```diff
@@ -47,20 +47,49 @@ struct ParsedSig {
     const TypeArgSpec &findTypeArgByName(core::NameRef name) const;
 };
 
+struct TypeSyntaxArgs {
+    bool allowSelfType = false;
+    bool allowRebind = false;
+    core::SymbolRef untypedBlame;
+
+    TypeSyntaxArgs(const bool allowSelfType, const bool allowRebind, core::SymbolRef untypedBlame)
+        : allowSelfType(allowSelfType), allowRebind(allowRebind), untypedBlame(untypedBlame) {}
+
+    static TypeSyntaxArgs forGetResultType(core::SymbolRef untypedBlame) {
+        return TypeSyntaxArgs(true, false, untypedBlame);
+    }
+
+    static TypeSyntaxArgs forParseSig(core::SymbolRef untypedBlame) {
+        return TypeSyntaxArgs(true, false, untypedBlame);
+    }
```

Comment:
Seems kind of weird to mention the names of implementation-specific methods in the public API of `TypeSyntaxArgs`. Thoughts? I wonder if it might be more readable if instead the API at the use site is something like

```cpp
auto allowSelfType = true;
auto allowRebind = false;
job.lhs.data(ctx)->resultType =
    TypeSyntax::getResultType(ctx, *(job.rhs), ParsedSig{}, TypeSyntaxArgs{allowSelfType, allowRebind, job.lhs});
```

Reply:
I'm OK with this. My hope in making these special-case constructor functions was that we could avoid situations where we have something like `TypeSyntaxArgs{bool,bool,val}`, which is equally difficult to read as just passing the flags through. If we name the intermediates ad you've suggested, that's not so bad.